agents/baseAgent.go

```diff
@@ -0,0 +1,63 @@
+package agents
+
+import (
+	"strings"
+
+	"github.com/tmc/langchaingo/chains"
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/tools"
+)
+
+type Agent struct {
+	Llm   llms.LLM
+	Chain chains.Chain
+	Tools []tools.Tool
+}
+
+type AgentInterface interface {
+	Run(input string) string
+}
+
+func FromLlmAndTools(llm llms.LLM, chain chains.Chain, tools []tools.Tool) Agent {
```

Comment:
constructors in go almost always are in the form of (package).NewXXX where XXX is the type in question.